contracts/RenExSettlement.sol

```diff
@@ -212,182 +247,216 @@ contract RenExSettlement is Ownable {
+            secondaryTokenVolume: secondaryTokenVolume,
+            priorityToken: priorityToken,
+            secondaryToken: secondaryToken,
+            priorityTokenAddress: priorityTokenAddress,
+            secondaryTokenAddress: secondaryTokenAddress,
+            timestamp: now
+        });
     }
 
-    function isEthereumBased(address _tokenAddress) private pure returns (bool) {
-        return (_tokenAddress != address(0x0));
+    /// @notice Order parity is set by the order tokens are listed. This returns
+    /// whether an order is a buy or a sell.
+    /// @return true if _orderID is a buy order.
+    function isBuyOrder(bytes32 _orderID) private view returns (bool) {
+        uint64 tokens = orderDetails[_orderID].tokens;
+        uint32 firstToken = uint32(tokens >> 32);
+        uint32 secondaryToken = uint32(tokens);
+        return (firstToken < secondaryToken);
     }
 
-    function subtractDarknodeFee(uint256 value) internal pure returns (uint256, uint256) {
+    /// @return (value - fee, fee) where fee is 0.2% of value
+    function subtractDarknodeFee(uint256 value) private pure returns (uint256, uint256) {
         uint256 newValue = (value * (DARKNODE_FEES_DENOMINATOR - DARKNODE_FEES_NUMERATOR)) / DARKNODE_FEES_DENOMINATOR;
         return (newValue, value - newValue);
     }
 
-    function getMatchDetails(bytes32 _orderID) public view returns (bytes32, bytes32, uint256, uint256, uint32, uint32) {
-        bytes32 matchingOrderID = orderbookContract.orderMatch(_orderID)[0];
-        bytes32 matchID;
-        MatchDetails memory details;
-        if (orderDetails[_orderID].parity == uint8(OrderParity.Buy)) {
-            matchID = keccak256(abi.encodePacked(_orderID, matchingOrderID));
-
-            details = matchDetails[matchID];
+    /// @return true if _tokenAddress is 0x0, representing a token that is not
+    /// on Ethereum
+    function isEthereumBased(address _tokenAddress) private pure returns (bool) {
+        return (_tokenAddress != address(0x0));
+    }
 
-            return (_orderID, matchingOrderID, details.highTokenVolume, details.lowTokenVolume, details.highToken, details.lowToken);
+    /// @notice Computes (numerator / denominator) * 10 ** scale
+    function joinFraction(uint256 numerator, uint256 denominator, int16 scale) private pure returns (uint256) {
+        if (scale >= 0) {
+            // Check that (10**scale) doesn't overflow
+            assert(scale <= 77); // log10(2**256) = 77.06
+            return numerator.mul(10 ** uint256(scale)) / denominator;
         } else {
-            matchID = keccak256(abi.encodePacked(matchingOrderID, _orderID));
-
-            details = matchDetails[matchID];
-
-            return (_orderID, matchingOrderID, details.lowTokenVolume, details.highTokenVolume, details.lowToken, details.highToken);
+            // joinFraction is only called with a minimum possible scale of -24
+            // assert((-scale) <= 77); // log10(2**256) = 77.06
```

Comment:
Why has this `assert` has been commented out?

Reply:
If scale is less than -77, it would overflow, but for now the smallest scale that can be passed in is -24 (when a token has 0 decimals and the price and volume offsets are 12). If `PRICE_OFFSET+VOLUME_OFFSET` is changed to be greater than 77, we can use the assert.

Reply:
It would definitely be worth leaving a comment alongside the commented out `assert` so that this is known.